### PR TITLE
updated summary sections for mcode generic summary section

### PR DESCRIPTION
**Pull request by gregquinn2001**
Addresses ASCODCP-1605. Created a generic cancer summary and the ability to have generic sections other than the overall default one. Uses entry element name to identify any CancerDisorder now.

View ccp endpoint mostly. All of MCODE represented now except adverse events and deceased (which don't exist in MCODE yet but are specified to eventually).

_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [x] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [x] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [ ] This pull request describes why these changes were made
- [ ] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by Dtphelan1** [comment]
DP1 - Attempting to edit the minimap layout throws the following error: image

File: src/summary/CoreCancerPilotSummaryMetadata.jsx [hunk 1]
```diff
@@ -1,50 +1,22 @@
...
-                                {clinicalEvent: X, defaultVisualizer: Y}                The specified visualizer type Y will be used if the current
-                                                                                        clinical event is X otherwise the first visualizer
-                                                                                        registered for the data type will be used.
-                                ["X", {clinicalEvent: Y, defaultVisualizer: Z}, ...]    A list of options allowing an overall default X that is used
-                                                                                        if one of the specific clinical events (e.g. Y) doesn't match
-                                                                                        the currently selected one. If a specific one matches, it
-                                                                                        uses the corresponding default visualizer (e.g. Z)
-*/
+//import SarcomaCoreCancerPilotMetadata from './metadata/SarcomaCoreCancerPilotMetadata';
```
Comment (Dtphelan1) [inline src/summary/CoreCancerPilotSummaryMetadata.jsx]: Should we remove this commented out code, as well as that on lines 6 and 16?

File: src/summary/metadata/DefaultCoreCancerPilotMetadata.jsx [hunk 2]
```diff
@@ -1,7 +1,13 @@
 import MetadataSection from "./MetadataSection";
+// import FluxCancerDisorder from "../../model/mcode/FluxCancerDisorder";
```
Comment (Dtphelan1) [inline src/summary/metadata/DefaultCoreCancerPilotMetadata.jsx]: Should we remove this commented out code, as well as that on lines: 3, 7-10?

File: src/summary/SummaryMetadata.jsx [hunk 3]
```diff
@@ -106,7 +100,8 @@ export default class SummaryMetadata {
                 return obj.getMetadata(preferencesManager, patient, condition, roleType, role, specialty);
             });
         });
-        this.addCachedMetadata(prioritizedKeyList[keyIndex], metadata);
+        // cache the metadata but only if condition is not null because we force default in that case
+//        if (prioritizedKeyList[keyIndex] !== 'default') this.addCachedMetadata(prioritizedKeyList[keyIndex], metadata);
```
Comment (Dtphelan1) [inline src/summary/SummaryMetadata.jsx]: Do we no longer cache metadata? Any reason why?

**Comment by gregquinn2001** [comment]
[DP2] Restore metadata caching functionality for targeted data panel

**Comment by gregquinn2001** [comment]
Fixed DP1 and DP2 and rebased

**Comment by gregquinn2001** [comment]
I took Dylan off since he's on vacation now!

**Comment by Dtphelan1** [comment]
What, up don’t think I can fly and review at the same time? :p.   
  

  
Hopefully my comments before leaving were helpful and got things moving in the right direction! All told the PR looks good to me with the exception of the comments I made. But getting another pair of eyes on it current hurt.   
  

  
  

  
  
>

**Comment by gregquinn2001** [comment]
Sorry Dylan! Feel free to approve! :) I just rebased.

**Comment by gregquinn2001** [comment]
[GQ1] Tests are failing now!

**Comment by gregquinn2001** [comment]
I removed GQ1. My mistake.

**Comment by jafeltra** [comment]
This looks good to me! I tested out CCP and pilot1 a bit. Both look good to me and the tests pass.

**Pull request merged**